Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean

```diff
@@ -187,4 +188,117 @@ lemma fromColumns_col0_isTotallyUnimodular_iff (A : Matrix m n R) :
   rw [← transpose_isTotallyUnimodular_iff, transpose_fromColumns, transpose_col,
     fromRows_row0_isTotallyUnimodular_iff, transpose_isTotallyUnimodular_iff]
 
+-- TODO maybe rename and certainly move
+lemma in_set_range_singType_cast_mul_in_set_range_singType_cast {a b : R}
+    (ha : a ∈ Set.range SignType.cast) (hb : b ∈ Set.range SignType.cast) :
+    a * b ∈ Set.range SignType.cast := by
+  sorry
+
+variable {R : Type*} [LinearOrderedCommRing R] -- TODO reorganize
+
+-- TODO maybe rename and certainly move
+lemma in_set_range_singType_cast_iff_abs (a : R) :
+    a ∈ Set.range SignType.cast ↔ |a| ∈ Set.range SignType.cast := by
+  sorry
+
+-- TODO maybe rename and certainly move; do not require `R` to be algebraic structure
+lemma fromBlocks_submatrix (A₁ : Matrix m n R) (A₂ : Matrix m' n' R)
+    {α : Type*} (f : α → m ⊕ m') (g : α → n ⊕ n') :
+    (fromBlocks A₁ 0 0 A₂).submatrix f g =
+    (Matrix.fromBlocks
+      (A₁.submatrix
+        ((·.val.snd) : { x₁ : α × m // f x₁.fst = Sum.inl x₁.snd } → m)
+        ((·.val.snd) : { y₁ : α × n // g y₁.fst = Sum.inl y₁.snd } → n)
+      ) 0 0
+      (A₂.submatrix
+        ((·.val.snd) : { x₂ : α × m' // f x₂.fst = Sum.inr x₂.snd } → m')
+        ((·.val.snd) : { y₂ : α × n' // g y₂.fst = Sum.inr y₂.snd } → n')
+      )
+     ).submatrix f.decomposeSum g.decomposeSum := by
+  rw [f.eq_comp_decomposeSum, g.eq_comp_decomposeSum, ←Matrix.submatrix_submatrix]
+  aesop
+
+attribute [-simp] Fintype.card_ofIsEmpty Fintype.card_ofSubsingleton -- major performance issue
```

Comment:
After the last change, it should not matter anymore whether they are simp lemmas.